src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -1,90 +1,346 @@
-            break;
+            return coordinateToLine(coordinates, properties, colorFunc, altitudeFunc);
         }
         case 'polygon': {
-            const indices = [];
-            /* eslint-disable guard-for-in */
-            for (const id in coordinates.featureVertices) {
-                const polygon = coordinates.featureVertices[id];
-                const contour = vertices.slice(polygon.offset * 3, (polygon.offset + polygon.count) * 3);
-                const triangles = Earcut(contour, null, 3);
-                for (const indice of triangles) {
-                    indices.push(polygon.offset + indice);
-                }
-            }
-            geometry.setIndex(new THREE.BufferAttribute(new Uint16Array(indices), 1));
-            result = new THREE.Mesh(geometry);
-            break;
+            return coordinateToPolygon(coordinates, properties, colorFunc, altitudeFunc);
         }
         default:
     }
+}
 
-    result.featureVertices = coordinates.featureVertices;
-
-    return result;
+function coordinatesToMeshExtruded(coordinates, properties, colorFunc, altitudeFunc, extrudeFunc) {
+    if (!coordinates) {
+        return;
+    }
+    switch (coordinates.type) {
+        case 'point': {
+            return coordinateToPoint(coordinates, properties, colorFunc, altitudeFunc);
+        }
+        case 'linestring': {
+            return coordinateToLine(coordinates, properties, colorFunc, altitudeFunc);
+        }
+        case 'polygon': {
+            return coordinateToPolygonExtruded(coordinates, properties, colorFunc, altitudeFunc, extrudeFunc);
+        }
+        default:
+    }
 }
 
-function featureToThree(feature) {
-    const mesh = coordinatesToMesh(feature.geometry);
+function featureToThree(feature, coordinateToMeshFunc, colorFunc, altitudeFunc, extrudeFunc) {
+    const mesh = coordinateToMeshFunc(feature.geometry, feature.properties, colorFunc, altitudeFunc, extrudeFunc);
     mesh.properties = feature.properties;
     return mesh;
 }
 
-function featureCollectionToThree(featureCollection) {
+function featureCollectionToThree(featureCollection, coordinateToMeshFunc, colorFunc, altitudeFunc, extrudeFunc) {
     const group = new THREE.Group();
     for (const geometry of featureCollection.geometries) {
-        group.add(coordinatesToMesh(geometry));
+        const properties = featureCollection.features;
+        group.add(coordinateToMeshFunc(geometry, properties, colorFunc, altitudeFunc, extrudeFunc));
     }
     group.features = featureCollection.features;
     return group;
 }
 
+function randomColor(/* properties */) {
+    const color = new THREE.Color();
+    color.setHex(Math.random() * 0xffffff);
+    return color;
+}
+
 export default {
-    convert(feature) {
-        if (feature.geometries) {
-            return featureCollectionToThree(feature);
-        } else {
-            return featureToThree(feature);
-        }
+
+    convertNoElevation() {
```

Comment:
IMHO, you could have a single entry point here: `Feature2Mesh.convert` with a single `options` parameter.
See https://github.com/iTowns/itowns/pull/519#discussion_r146487033 

Example usages:
```js
Feature2Mesh.convert({ color: new THREE.Color(0xff0000), altitude: myAltitudeFunction });
Feature2Mesh.convert();
Feature2Mesh.convert({ color: myColorFunction, altitude: 150, extrude: myExtrudeFn });
```

Reply:
ok, good idea.